cfe_internal/enterprise/federation/federation.cf

```diff
@@ -236,6 +248,35 @@ bundle agent transport_user
       "need to fix incorrect ssh context for transport user but restorecon path in $(sys.libdir)/paths.cf ($paths.restorecon) does not resolve";
 }
 
+bundle agent clean_when_off
+# @brief cleanup changes made for federated reporting on a feeder
+# NOTE: a superhub turned off by removing federation-config.json or setting
+# target_state=off will not revert schema changes.
+# federation_manage_files will always run regardless of off or not
+# so as to be prepared for enablement via Mission Portal UI
+{
+  vars:
+    "user" string => "$(cfengine_enterprise_federation:transport_user.user)";
+    "home" string => "$(cfengine_enterprise_federation:transport_user.home)"; 
+
+  users:
+    am_off::
+      "$(user)"  
+        policy => "absent";
+
+  files::
+    am_off::
+      "$(home)"
+      delete => tidy;
```

Comment:
I don't think this bundle needs to check `am_off`. It's does cleanup and knows *how*, it shouldn't care about *when*.